simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt

```diff
@@ -687,23 +686,54 @@ class CatalogFragmentFeed : Fragment() {
     search: FeedSearch?
   ) {
     val context = this.requireContext()
-    val toolbar = toolbarHost.findToolbar()
-    this.configureToolbarTitles(context, toolbar, ownership, title)
-    this.configureToolbarMenu(context, toolbar, search, title)
+    this.configureToolbarNavigation(context, toolbarHost, ownership)
+    this.configureToolbarTitles(context, toolbarHost, ownership, title)
+    this.configureToolbarMenu(context, toolbarHost, search, title)
+  }
 
-    toolbarHost.toolbarSetBackArrowConditionally(
-      context = context,
-      shouldArrowBePresent = { this.findNavigationController().backStackSize() > 1 },
-      onArrowClicked = { this.findNavigationController().popBackStack() })
+  @UiThread
+  private fun configureToolbarNavigation(
+    context: Context,
+    toolbarHost: ToolbarHostType,
+    ownership: CatalogFeedOwnership
+  ) {
+    val toolbar = toolbarHost.findToolbar()
+    try {
+      val navigationController = this.findNavigationController()
+      val isRoot = navigationController.backStackSize() == 1
+
+      if (isRoot) {
+        when (ownership) {
+          is OwnedByAccount -> {
+            toolbar.navigationIcon = context.getDrawable(R.drawable.accounts)
+            toolbar.navigationContentDescription = context.getString(R.string.catalogAccounts)
+            toolbar.setNavigationOnClickListener {
+              this.openAccountPickerDialog(ownership.accountId)
+            }
+          }
+          else -> toolbarHost.toolbarUnsetArrow()
+        }
+      } else {
+        toolbar.navigationIcon = toolbarHost.toolbarIconBackArrow(context)
+        toolbar.navigationContentDescription = null
+        toolbar.setNavigationOnClickListener { navigationController.popBackStack() }
+      }
+    } catch (e: Exception) {
+      // Note: The call to findNavigationController may throw an IllegalArgumentException.
```

Comment:
`BottomNavigator` sucks. :smile:

Reply:
lol, but that method is in our code 😅

Reply:
It's in our code, but it's written the way it is due to limitations imposed by `BottomNavigator` (we can't guarantee, at any given moment, that we actually have access to the navigator). There's like a one line change to guarantee that we always have access, but it causes sporadic fragment transaction exceptions if we make it. :roll_eyes: